test/TestInternalGrains/HashBasedPlacementGrain.cs

```diff
@@ -0,0 +1,9 @@
+﻿using Orleans.Placement;
+
+namespace UnitTests.Grains
+{
+    [HashBasedPlacement(true)]
```

Comment:
I was going to check if I can remove the test-only sorting now. I might have to change the test grains to return `SiloAddress` instead of a string to make sure client sorts silos the same way as they do themselves.